Src/ILGPU/IR/Location.cs

```diff
@@ -246,4 +265,101 @@ public class FileLocation : Location
 
         #endregion
     }
+
+    /// <summary>
+    /// A collection of locations than is manipulated as a stack.
+    /// </summary>
+    public class CompilationStackLocation : Location
+    {
+        #region Instance
+
+        private ImmutableStack<Location> Stack { get; }
```

Comment:
Nit: move properties (even private ones) to a `Properties` region.

Reply:
Fixed 👍